src/main/java/net/neoforged/neoforge/registries/DeferredRegister.java

```diff
@@ -580,6 +595,32 @@ protected <I extends Item> DeferredItem<I> createHolder(ResourceKey<? extends Re
         }
     }
 
+    /**
+     * Specialized DeferredRegister for {@link DataComponentType DataComponentTypes}.
+     */
+    public static class DataComponents extends DeferredRegister<DataComponentType<?>> {
+        protected DataComponents(String namespace) {
+            super(Registries.DATA_COMPONENT_TYPE, namespace);
+        }
+
+        /**
+         * Convenience method that constructs a builder for use in the operator. Use this to avoid inference issues.
+         * 
+         * @param name    The name for this data component type. It will automatically have the {@linkplain #getNamespace() namespace} prefixed.
+         * @param builder The unary operator, which is passed a new builder for user operations, then builds it when registered.
+         * @return A {@link DeferredDataComponentType} that will track updates from the registry for this type.
+         */
+        public <D> DeferredDataComponentType<D> registerBuilder(String name, UnaryOperator<DataComponentType.Builder<D>> builder) {
```

Comment:
the naming is not great - it's an actual component type that will be registered, not a builder

Reply:
The object that the user will ever interact with in this case is a builder. Hence the shortened name `registerBuilder` as opposed to something like `registerWithBuilder`. Though, if I think of another name or another name is suggested, I can update it

Reply:
could be `registerComponent`

Reply:
why not simply call it `register`, its already inside of a type specific register `DeferredRegister.DataComponents`.

`Blocks` and `Items` dont call theirs `registerItem` or `registerBlock` so why should this be any different?
and the places where they do, are only named differently cause parameters clash or its for a sub-type (item block)

Reply:
![image](https://github.com/neoforged/NeoForge/assets/13494793/216a0e6e-d58f-4286-acc4-1e6a92c3090e)

They do call them like that... The ones that are called `register` are covariant overrides from `DeferredRegister`.

Reply:
> why not simply call it `register`, its already inside of a type specific register `DeferredRegister.DataComponents`.
> 
> `Blocks` and `Items` dont call theirs `registerItem` or `registerBlock` so why should this be any different? and the places where they do, are only named differently cause parameters clash or its for a sub-type (item block)

In this case, it does clash, due to type erasure from `Function<RL, T>` and `UnaryFunction<T>` which extends `Function<T,T>`